src/ILCompiler.Compiler/src/Compiler/PreInitFieldInfo.cs

```diff
@@ -145,6 +150,10 @@ public static List<PreInitFieldInfo> GetPreInitFieldInfos(TypeDesc type)
                 if (!field.IsStatic)
                     continue;
 
+                bool isValueType = IsSupportedValueType(field.FieldType);
```

Comment:
Sounds good. Didn't know we had `HasGCStaticBase` - fixed. It also turned out I need to handle ThreadStatic case as well.